data/store/mongo/mongo.go

```diff
@@ -913,3 +925,273 @@ func validateAndTranslateSelectors(selectors *data.Selectors) (bson.M, error) {
+}
+
+func (d *DataRepository) GetLastUpdated(ctx context.Context, id string) (time.Time, error) {
+    var lastUpdated time.Time
+	if ctx == nil {
+		return lastUpdated, errors.New("context is missing")
+	}
+
+	if id == "" {
+		return lastUpdated, errors.New("id is missing")
+	}
+
+	var dataSet []*data.DataSet
+	selector := bson.M{
+		"_active": true,
+		"_userId": id,
+        "type": "cbg",
+	}
+
+	findOptions := options.Find()
+    findOptions.SetSort(bson.D{{"time", -1}})
+    findOptions.SetLimit(1)
+
+	cursor, err := d.Find(ctx, selector, findOptions)
+
+    if err != nil {
+		return lastUpdated, errors.Wrap(err, "unable to get last cbg date")
+	}
+
+	if err = cursor.All(ctx, &dataSet); err != nil {
+		return lastUpdated, errors.Wrap(err, "unable to decode last cbg date")
+	}
+
+    lastUpdated, err = time.Parse(time.RFC3339Nano, *dataSet[0].Time)
+
+	return lastUpdated, nil
+}
+
+func (s *Store) NewSummaryRepository() store.SummaryRepository {
+	return &SummaryRepository{
+		s.Store.GetRepository("summary"),
+	}
+}
+
+type SummaryRepository struct {
+	*storeStructuredMongo.Repository
+}
+
+func (d *SummaryRepository) EnsureIndexes() error {
+	return d.CreateAllIndexes(context.Background(), []mongo.IndexModel{
+		{
+			Keys: bson.D{
+				{Key: "_userId", Value: 1},
+			},
+			Options: options.Index().
+				SetBackground(true).
+				SetName("UserID"),
+		},
+        {
+			Keys: bson.D{
+				{Key: "lastUpdated", Value: 1},
+			},
+			Options: options.Index().
+				SetBackground(true).
+				SetName("LastUpdated"),
+		},
+	})
+}
+
+func (d *SummaryRepository) GetSummary(ctx context.Context, id string) (*data.Summary, error) {
+	if ctx == nil {
+		return nil, errors.New("context is missing")
+	}
+	if id == "" {
+		return nil, errors.New("summary UserID is missing")
+	}
+
+	//logger := log.LoggerFromContext(ctx).WithField("id", id)
+
+	var summary data.Summary
+	selector := bson.M{
+		"_userId": id,
+	}
+
+	err := d.FindOne(ctx, selector).Decode(&summary)
+
+    if err == mongo.ErrNoDocuments {
+		return nil, err
+	}
+
+	return &summary, err
+}
+
+func (d *SummaryRepository) UpdateSummary(ctx context.Context, summary *data.Summary) (*data.Summary, error) {
+	if ctx == nil {
+		return nil, errors.New("context is missing")
+	}
+	if summary == nil {
+		return nil, errors.New("summary object is missing")
+	}
+
+    if summary.UserID == "" {
+        return nil, errors.New("summary missing UserID")
+    }
+
+	opts := options.Replace().SetUpsert(true)
+    filter := bson.M{"_userId": summary.UserID}
+
+    _, err := d.ReplaceOne(ctx, filter, summary, opts)
+
+	//logger := log.LoggerFromContext(ctx).WithField("id", id)
+
+	return summary, err
+}
+
+func (d *SummaryRepository) GetAgedSummaries(ctx context.Context, minutes uint) ([]*data.Summary, error) {
+	if ctx == nil {
+		return nil, errors.New("context is missing")
+	}
+
+	//logger := log.LoggerFromContext(ctx).WithField("id", id)
+	startTime := time.Now().Add(time.Minute * -time.Duration(minutes))
```

Comment:
Sadly, no.
That results in a type error, an integer cannot be multiplied by time.Minutes. If it was a constant, it would allow it as constants are dynamically typed to Duration, but integers are not.